platform/sponge/src/main/java/me/lokka30/treasury/plugin/sponge/SpongeTreasuryPlugin.java

```diff
@@ -0,0 +1,125 @@
+package me.lokka30.treasury.plugin.sponge;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.Objects;
+import me.lokka30.treasury.plugin.core.Platform;
+import me.lokka30.treasury.plugin.core.TreasuryPlugin;
+import me.lokka30.treasury.plugin.core.config.ConfigAdapter;
+import me.lokka30.treasury.plugin.core.config.messaging.Messages;
+import me.lokka30.treasury.plugin.core.config.settings.Settings;
+import me.lokka30.treasury.plugin.core.logging.Logger;
+import me.lokka30.treasury.plugin.core.schedule.Scheduler;
+import me.lokka30.treasury.plugin.core.utils.PluginVersion;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import org.jetbrains.annotations.NotNull;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.scheduler.Task;
+
+public class SpongeTreasuryPlugin extends TreasuryPlugin implements Logger, Scheduler,
+        ConfigAdapter {
+
+    private final TreasurySponge plugin;
+    private final PluginVersion version;
+
+    private Messages messages;
+    private Settings settings;
+
+    private final File messagesFile;
+    private final File settingsFile;
+
+    public SpongeTreasuryPlugin(TreasurySponge plugin) {
+        this.plugin = Objects.requireNonNull(plugin, "plugin");
+        messagesFile = new File(plugin.getDataDir().toFile(), "messages.yml");
+        settingsFile = new File(plugin.getDataDir().toFile(), "settings.yml");
+
+        // todo: somehow parse back the container#metadata#version back into a string
+        this.version = new PluginVersion("", this);
+    }
+
+    @Override
+    public @NotNull PluginVersion getVersion() {
+        return this.version;
+    }
+
+    @Override
+    public @NotNull Platform platform() {
+        return Platform.SPONGE;
+    }
+
+    @Override
+    public @NotNull Path pluginsFolder() {
+        return null; // TODO
```

Comment:
To retrieve the plugins directory (mods) you need `Sponge.configManager().pluginConfig(pluginContainer).directory().getParent()`

Reply:
Thanks!